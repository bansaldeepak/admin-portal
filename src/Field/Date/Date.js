import React from "react";

import { makeStyles } from "@material-ui/core/styles";
import MomentUtils from "@date-io/moment";
import { MuiPickersUtilsProvider, DatePicker } from "@material-ui/pickers";
import InputAdornment from "@material-ui/core/InputAdornment";

// import { getInputProps } from '../lib';
const useStyles = makeStyles((theme) => ({
  textField: {
    minWidth: "100px",
  },
}));

const EnhancedDate = (props) => {
  const classes = useStyles();
  const {
    data,
    value,
    name,
    label,
    placeholder,
    openTo,
    format,
    required,
    disabled,
    disablePast,
    disableDate,
    disableFuture,
    helptext,
    min,
    max,
    readonly,
    prefix,
    suffix,
    title,
    maxlength,
    minlength,
    step,
    handleChange,
  } = props;

  // const { format } = props;
  // if (value !== undefined) {
  //   value = (format) ? value.format(format) : value.format('YYYY-MM-DD');
  // }
  // More properties handled in useEffect
  // format, disablePast, disableFuture, disableDate, openTo, min, max, readonly, maxlength, minlength, step

  return (
    <MuiPickersUtilsProvider utils={MomentUtils}>
      <DatePicker
        key={name}
        name={name}
        label={label}
        placeholder={placeholder}
        className={classes.textField}
        value={value !== undefined ? value : null}
        openTo={openTo !== undefined ? openTo : "date"}
        views={["year", "month", "date"]}
        onChange={(date) => handleChange(date)}
        margin="normal"
        format={format !== undefined ? format : "DD/MM/YYYY"}
        required={required}
        disabled={disabled}
        disablePast={disablePast !== undefined ? disablePast : false}
        shouldDisableDate={
          typeof disableDate === "function"
            ? (day) => {
                return disableDate(day, data);
              }
            : null
        }
        disableFuture={disableFuture !== undefined ? disableFuture : false}
        helperText={typeof helptext === "function" ? helptext(data) : helptext}
        minDate={typeof min === "function" ? min(data) : min}
        maxDate={typeof max === "function" ? max(data) : max}
        InputProps={{
          disabled: typeof disabled === "function" ? disabled(data) : disabled,
          readOnly: typeof readonly === "function" ? readonly(data) : readonly,
          startAdornment: prefix ? (
            <InputAdornment position="start">{prefix}</InputAdornment>
          ) : null,
          endAdornment: suffix ? (
            <InputAdornment position="end">{suffix}</InputAdornment>
          ) : null,
          inputProps: {
            title: typeof title === "function" ? title(data) : title,
            min: typeof min === "function" ? min(data) : min,
            max: typeof max === "function" ? max(data) : max,
            maxLength:
              typeof maxlength === "function" ? maxlength(data) : maxlength,
            minLength:
              typeof minlength === "function" ? minlength(data) : minlength,
            step: typeof step === "function" ? step(data) : step,
          },
        }}
      />
    </MuiPickersUtilsProvider>
  );
};

export default EnhancedDate;
