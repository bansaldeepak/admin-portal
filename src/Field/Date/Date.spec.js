import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";
import Date from "./Date";

configure({ adapter: new Adapter() });

describe("Date", () => {
  it("Render Date with default options", () => {
    const component = shallow(<Date name="date" />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
