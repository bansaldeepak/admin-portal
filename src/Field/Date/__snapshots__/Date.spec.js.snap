// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Render Date with default options 1`] = `
<MuiPickersUtilsProvider
  utils={[Function]}
>
  <PickerWithState
    InputProps={
      Object {
        "disabled": undefined,
        "endAdornment": null,
        "inputProps": Object {
          "max": undefined,
          "maxLength": undefined,
          "min": undefined,
          "minLength": undefined,
          "step": undefined,
          "title": undefined,
        },
        "readOnly": undefined,
        "startAdornment": null,
      }
    }
    allowKeyboardControl={true}
    className="makeStyles-textField-1"
    disableFuture={false}
    disablePast={false}
    format="DD/MM/YYYY"
    invalidDateMessage="Invalid Date Format"
    key="date"
    margin="normal"
    maxDate={2100-01-01T00:00:00.000Z}
    maxDateMessage="Date should not be after maximal date"
    minDate={1900-01-01T00:00:00.000Z}
    minDateMessage="Date should not be before minimal date"
    name="date"
    onChange={[Function]}
    openTo="date"
    shouldDisableDate={null}
    value={null}
    views={
      Array [
        "year",
        "month",
        "date",
      ]
    }
  />
</MuiPickersUtilsProvider>
`;
